app/src/main/java/ch/epfl/sdp/blindly/user/User.kt

```diff
@@ -0,0 +1,134 @@
+package ch.epfl.sdp.blindly.user
+
+import android.os.Build
+import android.util.Log
+import androidx.annotation.RequiresApi
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.ktx.getField
+import kotlinx.serialization.Serializable
+import java.time.LocalDate
+import java.time.Period
+
+/**
+ * A class used to represent a user
+ */
+@Serializable
+class User private constructor(
+                val username: String?,
+                val location: String?,
+                val birthday: String?,
+                val gender: String?,
+                val sexual_orientations: List<String>,
+                val show_me: String?,
+                val passions: List<String>,
+                val radius: Int?,
+                val matches: List<User>,
+                val description: String?) {
+
+    /**
+     * A builder used to partially initialize a user during the profile_setup activities
+     * Made serializable so that it can be put in a bundle and passed as extrax
+     */
+    @Serializable
+    data class Builder(
+        var username: String? = null,
+        var location: String? = null,
+        var birthday: String? = null,
+        var gender: String? = null,
+        var sexual_orientations: List<String> = listOf(),
+        var show_me: String? = null,
+        var passions: List<String> = listOf(),
+        var radius: Int? = null,
+        var matches: List<User> = listOf(),
+        var description: String? = null) {
```

Comment:
It might be better to use lateinit if you don't want your vars to be null again after being assigned.

Reply:
What do you mean by null again after being assign?

Reply:
I mean, is it useful for you to assign these variables to null after they have been assigned to a non-null argument ? If this is not the case, it is better to use lateinit. With lateinit, the variable is set to null until it is assigned, and cannot be reassigned to null.

Reply:
I don't believe they're reassigned to null after being assigned since there's all these setter functions that uses 
.apply{} on the same instance of builder each time